File: app/src/com/njust/helper/library/SearchSuggestionProvider.java
package com.njust.helper.library;

import android.content.SearchRecentSuggestionsProvider;

public class SearchSuggestionProvider extends SearchRecentSuggestionsProvider {

    public final static String AUTHORITY = "com.njust.helper.library.SearchSuggestionProvider";
    public final static int MODE = DATABASE_MODE_QUERIES;

    public SearchSuggestionProvider() {
        setupSuggestions(AUTHORITY, MODE);
    }
}

File: app/src/main/java/com/njust/helper/library/search/SearchSuggestionProvider.kt
package com.njust.helper.library.search

import android.content.SearchRecentSuggestionsProvider

class SearchSuggestionProvider : SearchRecentSuggestionsProvider() {
  init {
    setupSuggestions(AUTHORITY, MODE)
  }

  companion object {
    // For consistency, use old class name for Authority.
    const val AUTHORITY = "com.njust.helper.library.SearchSuggestionProvider"
    const val MODE = SearchRecentSuggestionsProvider.DATABASE_MODE_QUERIES
  }
}
